### PR TITLE
Support `label` attrib in shortcodes and gutenberg blocks

### PR DESCRIPTION
**Pull request by gordonsh24**
https://onthegosystems.myjetbrains.com/youtrack/issue/wpmltm-3848

### CODE REVIEW

**Comment by gordonsh24** [comment]
We can define labels in various way. Let's assume we have such block:

```
        <gutenberg-block type="toolset-blocks/button" translate="1">
            <xpath >//span</xpath>  
            <xpath >//a/@href</xpath>
        </gutenberg-block>
```

which defined a Toolset button. We have 2 fields to translate here: button label and URL. 

We can add one fallback label which will be applied for both fields:

```
        <gutenberg-block type="toolset-blocks/button" translate="1" label="Button label">
            <xpath >//span</xpath>  
            <xpath >//a/@href</xpath>
        </gutenberg-block>
```

We can also define individual labels for both fields:

```
        <gutenberg-block type="toolset-blocks/button" translate="1">
            <xpath label="Button label">//span</xpath>  
            <xpath label="Button url">//a/@href</xpath>
        </gutenberg-block>
```

We can specify individual label for URL but use a fallback for button's label:

```
        <gutenberg-block type="toolset-blocks/button" translate="1" label="Button">
            <xpath >//span</xpath>  
            <xpath label="Button url">//a/@href</xpath>
        </gutenberg-block>
```

The same mechanism applies for `key` node:

```
        <gutenberg-block type="toolset-blocks/button" translate="1">  
            <key label="Button label" name="buttonText" />
            <key name="url" label="Button url" />
        </gutenberg-block>
```

**Comment by gordonsh24** [comment]
@OnTheGoSystems/wpml @OnTheGoSystems/compatibility Please review it

**Review by andreasciamanna — DISMISSED** [review]

File: src/strings-in-block/class-attributes.php [hunk 1]
```diff
@@ -43,12 +43,13 @@ private function findStringsRecursively( array $attrs, array $config_keys, $bloc
...
 					$strings,
-					$this->findStringsRecursively( $attr_value, $children_config_keys, $block_name )
+					$this->findStringsRecursively( $attr_value, $children_config_keys, $block )
 				);
 			} elseif ( ! is_numeric( $attr_value ) ) {
 				$type      = self::get_string_type( $attr_value );
-				$string_id = $this->get_string_id( $block_name, $attr_value );
-				$strings[] = $this->build_string( $string_id, $block_name, $attr_value, $type );
+				$string_id = $this->get_string_id( $block->blockName, $attr_value );
+				$label = isset( $config_keys[ $attr_key ]['label'] ) ? $config_keys[ $attr_key ]['label'] : $this->get_block_label( $block );
```
Comment (strategio) [inline src/strings-in-block/class-attributes.php]: Alignment @gordonsh24

**Review by strategio — DISMISSED** [review]
Looks good @gordonsh24 +1

**Review by strategio — APPROVED** [review]
+1 @gordonsh24

**Review by brucepearson — APPROVED** [review]

**Pull request merged**